src/utils.jl

```diff
@@ -66,3 +66,88 @@ function check_pkg_version(pkg::String, ver::String;
            pkg_info[pkg].version >= VersionNumber(ver) :
            pkg_info[pkg].version > VersionNumber(ver)
 end
+
+
+# RetCode handling for BBO and others.
+using Logging
+ # Define the ReturnCode type using @enum
+@enum ReturnCode begin
```

Comment:
These already exist in SciMLBase right, you don't need to define this again here